src/common.rs

```diff
@@ -109,91 +109,47 @@ pub(crate) fn config_file_get() -> String {
 
 /// Returns revocation ip from keylime.conf if env var not present
 pub(crate) fn revocation_ip_get() -> Result<String> {
-    match env::var("REVOCATION_IP") {
-        Ok(ip) => {
-            // The variable length must be larger than 0 to accept
-            if !ip.is_empty() {
-                Ok(ip)
-            } else {
-                config_get("general", "receive_revocation_ip")
-            }
-        }
-        _ => config_get("general", "receive_revocation_ip"),
-    }
+    config_get_env("general", "receive_revocation_ip", "REVOCATION_IP")
 }
 
 /// Returns revocation port from keylime.conf if env var not present
 pub(crate) fn revocation_port_get() -> Result<String> {
-    match env::var("REVOCATION_PORT") {
-        Ok(ip) => {
-            // The variable length must be larger than 0 to accept
-            if !ip.is_empty() {
-                Ok(ip)
-            } else {
-                config_get("general", "receive_revocation_port")
-            }
-        }
-        _ => config_get("general", "receive_revocation_port"),
-    }
+    config_get_env("general", "receive_revocation_port", "REVOCATION_PORT")
 }
 
 /// Returns cloud agent IP from keylime.conf if env var not present
 pub(crate) fn cloudagent_ip_get() -> Result<String> {
-    match env::var("CLOUDAGENT_IP") {
-        Ok(ip) => {
-            // The variable length must be larger than 0 to accept
-            if !ip.is_empty() {
-                Ok(ip)
-            } else {
-                config_get("cloud_agent", "cloudagent_ip")
-            }
-        }
-        _ => config_get("cloud_agent", "cloudagent_ip"),
-    }
+    config_get_env("cloud_agent", "cloudagent_ip", "CLOUDAGENT_IP")
 }
 
 /// Returns cloud agent port from keylime.conf if env var not present
 pub(crate) fn cloudagent_port_get() -> Result<String> {
-    match env::var("CLOUDAGENT_PORT") {
-        Ok(ip) => {
-            // The variable length must be larger than 0 to accept
-            if !ip.is_empty() {
-                Ok(ip)
-            } else {
-                config_get("cloud_agent", "cloudagent_port")
-            }
-        }
-        _ => config_get("cloud_agent", "cloudagent_port"),
-    }
+    config_get_env("cloud_agent", "cloudagent_port", "CLOUDAGENT_PORT")
 }
 
 /// Returns registrar IP from keylime.conf if env var not present
 pub(crate) fn registrar_ip_get() -> Result<String> {
-    match env::var("REGISTRAR_IP") {
-        Ok(ip) => {
-            // The variable length must be larger than 0 to accept
-            if !ip.is_empty() {
-                Ok(ip)
-            } else {
-                config_get("registrar", "registrar_ip")
-            }
-        }
-        _ => config_get("registrar", "registrar_ip"),
-    }
+    config_get_env("registrar", "registrar_ip", "REGISTRAR_IP")
```

Comment:
this needs to be cloud_agent:

`config_get_env("cloud_agent", "registrar_ip", "REGISTRAR_IP")`

Reply:
Yes, this should be `cloud_agent` but I've just kept the old behaviour. 
If I change that the `keylime.conf` which currently doesn't contain that key and `registrar_port_get(...)` should be also updated.

Reply:
@lukehinds should I include the update of the `keylime.conf` in this PR or should I move those changes into a new PR?

Reply:
good question, at some point we need to think about how we insure parity between the config here and the config over n keylime/keylime.

I am fine with it being in this PR.

Reply:
Done. I've now updated the keylime.conf and moved the keys to the correct section.